### PR TITLE
Add span field meta_struct

### PR DESCRIPTION
**Pull request by estringana**
### Description

Add new field `meta_struct` to the span. This new field is a map where its values are encoded to msgpack and then the resulting codification to msgpack binary

APPSEC-52612

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: ext/serializer.c [hunk 1]
```diff
@@ -1017,6 +1045,27 @@ static void dd_serialize_array_metrics_recursively(zend_array *target, zend_stri
...
+    mpack_writer_destroy(&writer);
+
+    if (size == 0 || result == 0) {
+        return;
+    }
+
+    zval serialised;
+    ZVAL_STRINGL(&serialised, data, size);
+
+    dd_serialize_array_recursively(target, str, &serialised, false);
```
Comment (bwoebi) [inline ext/serializer.c:1064]: So, `serialised` is always a string, the only thing dd_serialize_array_recursively does ... is then adding it to the hashtable - it was confusing to read, so just simplify it:
```suggestion
    zend_hash_update(target, str, &serialised);
```

File: ext/serializer.c [hunk 2]
```diff
@@ -1017,6 +1045,27 @@ static void dd_serialize_array_metrics_recursively(zend_array *target, zend_stri
...
+    free(data);
+    zval_ptr_dtor(&serialised);
```
Comment (bwoebi) [inline ext/serializer.c:1066]:
```suggestion
```

And remove this now given that zend_hash_update() doesn't incref (the ddtrace_convert_to_string in dd_serialize_array_recursively was doing that though).

Comment (estringana) [reply ext/serializer.c]: Done! Thanks

**Review by bwoebi — APPROVED** [review]
Looks correct to me

**Review by Anilm3 — APPROVED** [review]
lgtm

**Pull request merged**